algo/src/main/java/org/neo4j/graphalgo/impl/MSBFSAllShortestPaths.java

```diff
@@ -0,0 +1,146 @@
+package org.neo4j.graphalgo.impl;
+
+import com.carrotsearch.hppc.AbstractIterator;
+import org.neo4j.graphalgo.api.Graph;
+import org.neo4j.graphalgo.impl.msbfs.MultiSourceBFS;
+import org.neo4j.graphdb.Direction;
+
+import java.util.Iterator;
+import java.util.Spliterators;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * AllShortestPaths:
```

Comment:
JavaDoc and Comments are still copy&pasted from the PD-ASP and need to be updated.